src/main/java/org/apache/commons/lang3/StringUtils.java

```diff
@@ -9508,20 +9514,27 @@ public static String wrapIfMissing(final String str, final char wrapWith) {
      * @param str
      *            the string to be wrapped, may be {@code null}
      * @param wrapWith
-     *            the char that will wrap {@code str}
+     *            the string that will wrap {@code str}
      * @return the wrapped string, or {@code null} if {@code str==null}
      * @since 3.5
      */
     public static String wrapIfMissing(final String str, final String wrapWith) {
         if (isEmpty(str) || isEmpty(wrapWith)) {
             return str;
         }
+        
+        final boolean wrapStart = !str.startsWith(wrapWith);
+        final boolean wrapEnd = !str.endsWith(wrapWith);
+        if (!wrapStart && !wrapEnd) {
+            return str;
+        }
+        
         final StringBuilder builder = new StringBuilder(str.length() + wrapWith.length() + wrapWith.length());
```

Comment:
I don't know if there is a practical reason to create `StringBuilder` with exact capacity, because it `str` can start with `wrapWith` but not end with it and vice versa.

Reply:
Yeah, looks like this is the maximum capacity, not necessarily the exact capacity we expect to have. Probably harmless/fine?

Reply:
Since the behavior has changed, I would expect a clarification in the Javadoc. Plus a matching test.

Reply:
@garydgregory Sure, that makes sense. Done.